pkg/kubelet/kubelet.go

```diff
@@ -995,6 +995,30 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *api.ObjectFieldSelector, pod
 	return fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
 }
 
+func (kl *Kubelet) lookupDnsService() (string, string, error) {
+	// Lookup the cluster DNS service and returns its info
+	if kl.serviceLister == nil {
+		return "", "", nil
+	}
+	services, err := kl.serviceLister.List()
+	if err != nil {
+		return "", "", fmt.Errorf("failed to list services when setting up env vars.")
+	}
+
+	for _, service := range services.Items {
+		if service.Name == "kubernetes-dns" && service.Namespace == "default" {
```

Comment:
Since this is a new thing, we get to define what it's called. Eric's suggestion was to use 'kubernetes-dns' as is fits in with the other 'kubernetes' (and the now expired 'kubernetes-ro') service.